icu4c/source/common/unicode/locid.h

```diff
@@ -362,6 +367,55 @@ class U_COMMON_API Locale : public UObject {
                                      UErrorCode&   success);
 #endif  /* U_HIDE_SYSTEM_API */
 
+#ifndef U_HIDE_DRAFT_API
+    /**
+     * Returns a Locale for the specified BCP47 language tag string.
+     * If the specified language tag contains any ill-formed subtags,
+     * the first such subtag and all following subtags are ignored.
+     * <p>
+     * This implements the 'Language-Tag' production of BCP47, and so
+     * supports grandfathered (regular and irregular) as well as private
+     * use language tags.  Private use tags are represented as 'x-whatever',
+     * and grandfathered tags are converted to their canonical replacements
+     * where they exist.  Note that a few grandfathered tags have no modern
+     * replacement, these will be converted using the fallback described in
+     * the first paragraph, so some information might be lost.
+     * @param tag     the input BCP47 language tag.
+     * @param status  error information if creating the Locale failed.
+     * @return        the Locale for the specified BCP47 language tag.
+     * @draft ICU 63
+     */
+    static Locale U_EXPORT2 forLanguageTag(StringPiece tag, UErrorCode& status);
+
+    /**
+     * Returns a well-formed language tag for this Locale.
+     * <p>
```

Comment:
Please document that when this function fails, partial output may have been written to the sink.

Your current implementation won't do that, but I think we want the freedom to write subtags incrementally rather than allocate and reallocate one big buffer.

Reply:
Done.